pkg/server/deploy/deploy.go

```diff
@@ -113,6 +108,10 @@ func (ops *DeployOperations) Deploy(ctx context.Context, user *database.User, ap
 		if !app.IsCronJob(a.ProcessType) {
 			ops.watchDeploy(appName, deployId, w, errChan)
 		}
+
+		if err := ops.appOps.SaveApp(a, user.Email); err != nil {
```

Comment:
Wouldn't it be better to save the app before watching the deploy?

Reply:
you're right, I'll change